### ISSUE TITLE
Can you please add the 3 Returning Variables: %stdout, %stderr, %err, like the "Run Shell" action has?

### ISSUE THREAD

**Issue by spaceman5** [termux/termux-tasker#21]
Hi

May I suggest to add 3 Returning Variables, that will be filled by the Termux Task plugin, after returning from the execution?

The needed 3 variables are:
%stdout
%stderr
%err (the returning error code from the program that was run)

Please see this screenshot:
https://i.postimg.cc/QdwwXNVX/Screenshot-20190417-144859-resize.png

(the first 2 appear, and the 3rd one does not appear, but Tasker fills it into a variable named %err)

That way we can easily get the Result from the program we ran.

Thank you very much

**Comment by Surrogard** [comment]
+1 from me
I use the plugin with automate and the only way currently to get things back is via broadcast or external file which works but is quite annoying to set up.

**Comment by ieatacid** [comment]
I concur on the usefulness of this request.

**Comment by Archenoth** [comment]
Oh, agreed!

The ability to allow returning data from Termux into Tasker would mean that you can do things like tell if your scripts actually succeed, let you get processed data directly into Tasker tasks, or return state that only Termux can see.

So this comment isn't basically just a "+1", I'll try to explain the impact this has; without this functionality, the ways to return data to Tasker involve either:

### Write files to Tasker-readable locations which
Since it's easier to explain messiness by showing what things look like practically, let's dive into creating a scene with three checkboxes that are set with simple checks in Termux!
![Service control popup with a few checkboxes indicating if certain services have started or not](https://user-images.githubusercontent.com/1385527/69573163-d1e78e80-0f82-11ea-95ab-317eda80a7f7.png)

First, to get the status of each service, you need to either write a new script, or add a hack to an existing script to write to a folder that Tasker can read. In this case I opted for [a hack in my service script](https://gist.github.com/Archenoth/25fb168edd899bd5398b18a65c280664/7d47c8e6245a09b2b37204637e549f8997fcad50#file-service-L14).

Then after implementing that, in order to read the result, the Task itself needs to perform the checks for each file (Easier in this case), or extract all the data from one file:
![The Tasker Task to open the above box, and set the state of the checkboxes](https://user-images.githubusercontent.com/1385527/69572735-07d84300-0f82-11ea-8bcc-b6c0001d26db.png)

The end result is, voila! I have 3 checkboxes on this scene that are checked based on if a service is started or not!

So, while it's not impossible, it's quite messy. Almost all of the mess would be removed by the ability to read the status code of tasks, even without stdout and stderr output.

### Send an intent
This requires setting up a new Tasker profile for every single variable you wish to pass back that you wish to do something based on. So if I wanted to create a Task that ran a command, and if the result matched something to run another Task, that would involve:

1) Setting up the Task to run the script
2) Writing the script which would do the check, and then run the shell command:
```bash
am broadcast --user 0 -a net.dinglish.tasker.the-thing-I-want-to-send -e message "$something"
```
3) Make a Tasker profile to react to the `net.dinglish.tasker.the-thing-I-want-to-send` intent and set it to a global variable, or to do the rest of the Task started in the Task started in Step 1

You would need to do this for every single communication between Tasker and Termux--so for something with 5 checks, that would involve making at least 6 profiles that do nothing but react to intents to continue your logic.

So again, while this is not impossible, it's also very messy.

Anyway, hope this explains what we are up against here..! I think this suggestion would be highly appreciated, so I wrote this up in hopes to give it a little more than just a simple "me too".

But even if this suggestion is declined, thanks for the project! It's extraordinarily useful~

**Comment by spaceman5** [comment]
I am glad other people need it too..
I hope the creators of Termux will notice this thread, and update the Termux Tasker plugin..

**Comment by Archenoth** [comment]
Hm, looking into it, it seems like adding this requires a bit more work than just updating this plugin.

Specifically, it looks like it requires an update in [the main app](https://github.com/termux/termux-app) to return those three values from the Service that handles the Intent [here.](https://github.com/termux/termux-app/blob/f4db9ff212a672572c2a8fbf2f494e67d38a6d27/app/src/main/java/com/termux/app/TermuxService.java#L143)

I'm not really an Android dev, so I'm not sure about the best way to do that.

**Comment by verboze** [comment]
I look forward to seeing Archenoth's PRs merged! I started offloading more complex Tasker logic to Termux, and it is becoming apparent how much this feature is needed.

**Comment by fornwall** [comment]
The PR of @Archenoth (great job!) has been merged and released in version 0.4 of the Termux:Tasker app.

Note that this also requires version 0.86 of the main Termux app - this will be released within a day.

**Comment by Archenoth** [comment]
Yays! I'm happy I could help~
